src/preamble.js

```diff
@@ -449,9 +449,8 @@ function UTF8ArrayToString(u8Array, idx) {
 #endif // TEXTDECODER
 
 #if TEXTDECODER == 2
-  return UTF8Decoder.decode(
-    u8Array.subarray ? u8Array.subarray(idx, endPtr) : new Uint8Array(u8Array.slice(idx, endPtr))
-  );
+  // Avoid redundant temp garbage allocation on empty strings by checking (endPtr > idx) first
```

Comment:
are empty strings common enough to be worth the extra code size here?

Reply:
Yeah, I was pondering that for a long while as well, but in the end leaned towards thinking it is, because the code path would otherwise do a temp garbage allocation also on empty strings in that .subarray() (or uint8array()+slice()). (subarray + js string). This can be beneficial for APIs that have optional string parameters where one can pass in a null pointer for a string that does not need to be used for the function call - such APIs won't then do redundant garbage when marshalling no-strings.